src/router/resolvers/index.ts

```diff
@@ -1551,11 +1551,27 @@ export const IGCESupportingDocumentationResolver = (current: string): string =>
   }
 };
 
+export const RFDResolver = (): string => {
+  if (!isDitcoUser()) {
+    return routeNames.RFD
+  }
+  return Summary.hasCurrentStepBeenVisited
+    ? routeNames.SummaryStepEight
+    : routeNames.CurrentlyHasFunding
+}
+
 export const CurrentlyHasFundingResolver = (current: string): string => {
+  const doesNotNeedFundingDoc = AcquisitionPackage.acquisitionPackage
+    ?.contracting_shop_require_funding_documents_for_submission_of_package === 'NO'
+
+  if (current === routeNames.RFD && doesNotNeedFundingDoc) {
+    return routeNames.SummaryStepEight
+  }
+
   return Summary.hasCurrentStepBeenVisited
     ? routeNames.SummaryStepEight
-    : isDitcoUser()
-      ? routeNames.CurrentlyHasFunding
+    : !isDitcoUser() ?
+      routeNames.CurrentlyHasFunding

```

Comment:
use `src/steps/10-FinancialDetails/CurrentlyHasFunding.spec.ts` as a reference, you'll find the code below to mock ditco user

`import * as acqPackageExportedFunctions from "@/store/acquisitionPackage";`

`jest.spyOn(acqPackageExportedFunctions, "isDitcoUser").mockReturnValue(false);`